app/views/parsers/_parser.html.erb

```diff
@@ -162,7 +162,7 @@
   <%= custom_form_with model: @parser, local: true do |f| %>
     <%= f.label :partner, 'Contributor' %>
     <%= f.select :partner, Partner.all.sort(name: 1).map { |p| [p.name, p.name] }, selected: @parser.partner.name %>
-    <%= f.label :source, 'Data source' %>
+    <%= f.label :source, 'Data source', data: { 'current_source_id': @parser.source_id } %>
```

Comment:
I think you can avoid this javascript by setting the `value` on the `grouped_collection_select`. No?

Reply:
na it's kind of strange but the value is currently set, but when the javascript that appends and removes options runs we lose the value. Even though the tag still has selected on it

Reply:
if you don't run this JS at all, the form works correctly it just shows all of the options

Reply:
Ah yeah, I thought it was on the first display. All good then :)

Reply:
😄